### PR TITLE
Firewall tests & fixes

### PR DESCRIPTION
**Pull request by outscale-jju**

### CODE REVIEW

File: src/firewall.c [hunk 1]
```diff
@@ -165,6 +164,13 @@ int pg_firewall_reload(struct pg_brick *brick, struct pg_error **errp)
 	return 0;
 }
 
+int pg_firewall_reload(struct pg_brick *brick, struct pg_error **errp)
+{
+	struct pg_firewall_state *state =
+		pg_brick_get_state(brick, struct pg_firewall_state);
+	return firewall_reload_internal(state, errp);
```
Comment (outscale-mgo) [inline src/firewall.c]: I would just do:
`return firewall_reload_internal(pg_brick_get_state(brick, struct pg_firewall_state));`

File: tests/firewall/tests.c [hunk 2]
```diff
@@ -162,6 +162,47 @@ static struct rte_mbuf *build_non_ip_packet(void)
...
+
+	/* let's burst ! */
+	pg_brick_poll(gen, &packet_count, &error);
+	g_assert(!error);
+
+	/* check collect brick */
+	pg_brick_west_burst_get(col, &filtered_mask, &error);
+	g_assert(!error);
+
+	return filtered_mask;
```
Comment (outscale-mgo) [inline tests/firewall/tests.c]: well, your missing some free here :)
struct pg_brick *gen, *fw, *col are all leaking

File: tests/firewall/tests.c [hunk 3]
```diff
@@ -162,6 +162,47 @@ static struct rte_mbuf *build_non_ip_packet(void)
 	return pkt;
 }
 
+/* send burst, get mask after firewall */
+static uint64_t firewall_scenario_filter(const char *rule,
+					 struct rte_mbuf **pkts,
+					 uint64_t mask)
```
Comment (outscale-mgo) [inline tests/firewall/tests.c]: the indentation doesn't look good on github

Comment (outscale-jju) [inline tests/firewall/tests.c]: As the norm is a mix of tabs and spaces, I would say that's normal

**Comment by outscale-jju** [comment]
merged

**Pull request closed without merging**